src/index.spec.ts

```diff
@@ -20,21 +23,41 @@ describe("json-rpc-e2e", () => {
       port: 8080
     });
 
+    client = new JSONRPCClient("http://localhost:8080/rpc/v1");
+
+    service = client.getService<TestService>("test");
+
     app = moduleRef.createNestMicroservice({ strategy: server });
     await new Promise(resolve => app.listen(resolve));
   });
 
-  it(`/rpc/v1/ test.invoke (POST)`, () => {
-    return request(server.server)
-      .post("/rpc/v1")
-      .send({ method: "test.invoke", params: { data: "hi" } })
-      .expect(200)
-      .expect({
-        data: "hi"
-      });
+  it(`should make and RPC call with the JSONRPCClient`, () => {
+    return service
+      .invokeClientService({ data: "hi" })
+      .then(res => expect(res.result.data).toStrictEqual({ data: "hi" }));
+  });
+
+  it(`should check error object properties from JSONRPCClient call`, () => {
+    const jsonRpcErrorObj = {
+      id: expect.stringMatching,
+      jsonrpc: "",
+      error: {
+        message: "",
+        code: 403,
+        data: {
+          fromService: "",
+          params: { data: "" }
+        }
+      }
+    };
+    return service.testError({ data: "hi" }).then(res => {
```

Comment:
The client should automatically unwrap the content. You should not be aware of any json-rpc implementation detail here.